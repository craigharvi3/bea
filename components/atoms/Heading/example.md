Headings

```js
<div>
  <Heading level={1}>Heading</Heading>
  <Heading level={2}>Heading</Heading>
  <Heading level={3}>Heading</Heading>
  <Heading level={4}>Heading</Heading>
  <Heading level={5}>Heading</Heading>
</div>
```

