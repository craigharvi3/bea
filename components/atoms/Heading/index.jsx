// TODO: This is not production ready
import React from 'react';
import PropTypes from 'prop-types';

const Heading = ({ level, children, ...props }) =>
  React.createElement(`h${level}`, props, children);

Heading.propTypes = {
  level: PropTypes.number.isRequired,
  children: PropTypes.node
};

export default Heading;
