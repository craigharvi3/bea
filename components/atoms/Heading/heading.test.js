import React from 'react';
import { shallow } from 'enzyme';
import Heading from './';

const props = {
  level: 1
};

it('should render a heading', () => {
  const actual = shallow(<Heading {...props}>Heading</Heading>);
  expect(actual).toMatchSnapshot();
});
