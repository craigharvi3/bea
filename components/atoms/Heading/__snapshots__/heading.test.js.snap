// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a heading 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Heading
    level={1}
  >
    Heading
  </Heading>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": "Heading",
    },
    "ref": null,
    "rendered": "Heading",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "Heading",
      },
      "ref": null,
      "rendered": "Heading",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
